### PR TITLE
Enable offloading attention and layer / RMS norm to CUTLASS

### PR DESCRIPTION
**Pull request by masahi**
With TVM built with `USE_CUTLASS`, adding `--use-cutlass-attn --use-cutlass-norm` enables offloading attention and layer / RMS norm to CUTLASS.

On RTX 4080 and Vicuna 7B, attention offload brings 12-15 tok / s improvement, and RMS norm offload adds another 4-5 tok / sec.

### CODE REVIEW

File: mlc_llm/core.py [hunk 1]
```diff
@@ -126,6 +130,26 @@ class BuildArgs:
...
+    use_cutlass_norm: bool = field(
+        default=False,
+        metadata={
+            "help": (
+                "Offload layer and RMS norm operations to CUTLASS when the target is CUDA"
+                "and TVM has been built with CUTLASS enabled."
+            ),
+            "action": "store_true",
+        },
+    )
```
Comment (junrushao) [inline mlc_llm/core.py:152]: can we merge them into one config: `use_cutlass="norm;attn"`?

Comment (masahi) [reply mlc_llm/core.py:152]: I want to stick to the existing convention to use boolean flags for this PR.

Comment (junrushao) [reply mlc_llm/core.py:152]: Will be good future work to do

**Comment by junrushao** [comment]
Super excited to have this upstreamed!

**Review by junrushao — APPROVED** [review]

**Pull request merged**